src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -126,134 +148,296 @@ private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISc
+            private static ITensorValueGetter[] GetTensorValueGetters(IRow input,
+                int[] inputColIndices,
+                bool[] isInputVector,
+                DataType[] onnxInputTypes,
+                OnnxShape[] onnxInputShapes)
+            {
+                var srcTensorGetters = new ITensorValueGetter[inputColIndices.Length];
+                for (int i = 0; i < inputColIndices.Length; i++)
+                {
+                    int colIndex = inputColIndices[i];
+                    srcTensorGetters[i] = CreateTensorValueGetter(input, onnxInputTypes[i], isInputVector[i], colIndex, onnxInputShapes[i]);
+                }
+                return srcTensorGetters;
+            }
+
+            private static ITensorValueGetter CreateTensorValueGetter(IRow input, DataType onnxType, bool isVector, int colIndex, OnnxShape onnxShape)
+            {
+                var type = OnnxUtils.OnnxToMlNetType(onnxType).RawType;
+                Contracts.AssertValue(type);
+                return Utils.MarshalInvoke(CreateTensorValueGetter<int>, type, input, isVector, colIndex, onnxShape);
+            }
+
+            private static ITensorValueGetter CreateTensorValueGetter<T>(IRow input, bool isVector, int colIndex, OnnxShape onnxShape)
+            {
+                if (isVector)
+                    return new TensorValueGetterVec<T>(input, colIndex, onnxShape);
+                return new TensorValueGetter<T>(input, colIndex);
+            }
+
+            private class TensorValueGetter<T> : ITensorValueGetter
+            {
+                private readonly ValueGetter<T> _srcgetter;
+
+                public TensorValueGetter(IRow input, int colIndex)
+                {
+                    _srcgetter = input.GetGetter<T>(colIndex);
+                }
+                public Tensor GetTensor()
+                {
+                    var scalar = default(T);
+                    _srcgetter(ref scalar);
+                    return OnnxUtils.CreateScalarTensor(scalar);
+                }
+            }
+
+            private class TensorValueGetterVec<T> : ITensorValueGetter
+            {
+                private readonly ValueGetter<VBuffer<T>> _srcgetter;
+                private readonly OnnxShape _tensorShape;
+                private VBuffer<T> _vBuffer;
+                private VBuffer<T> _vBufferDense;
+                public TensorValueGetterVec(IRow input, int colIndex, OnnxShape tensorShape)
+                {
+                    _srcgetter = input.GetGetter<VBuffer<T>>(colIndex);
+                    _tensorShape = tensorShape;
+                    _vBuffer = default;
+                    _vBufferDense = default;
+                }
+                public Tensor GetTensor()
+                {
+                    _srcgetter(ref _vBuffer);
+                    _vBuffer.CopyToDense(ref _vBufferDense);
+                    return OnnxUtils.CreateTensor(_vBufferDense.GetValues().ToArray(), _tensorShape);
```

Comment:
>_vBufferDense.GetValues().ToArray() [](start = 50, length = 35)

I would probably change CreateTensor to accept ReadonlySpan<T> than cast every span to array.

Span is our new reality with VBuffers, you shouldn't expect arrays anymore.